src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -830,15 +833,525 @@ theorem contains_keys [EquivBEq α] [LawfulHashable α] (h : m.1.WF) {k : α} :
+  simp_to_model using getKey_insertListIfNewUnit_of_mem_of_contains_eq_false
+
+theorem getKey_insertManyIfNewUnit_list_mem_of_mem_of_contains [EquivBEq α] [LawfulHashable α]
+    (h : m.1.WF) {l : List α} {k : α}
+    (distinct : l.Pairwise (fun a b => (a == b) = false))
+    (h': l.contains k) {mem' : m.contains k} {h'} :
+    getKey (insertManyIfNewUnit m l).1 k h' = getKey m k mem' := by
+  simp_to_model using getKey_insertListIfNewUnit_of_mem_of_contains
+
+theorem getKey!_insertManyIfNewUnit_list_of_contains_eq_false [EquivBEq α] [LawfulHashable α]
+    [Inhabited α] (h : m.1.WF) {l : List α} {k : α}
+    (h' : l.contains k = false) :
+    getKey! (insertManyIfNewUnit m l).1 k = getKey! m k := by
+  simp_to_model using getKey!_insertListIfNewUnit_of_contains_eq_false
+
+theorem getKey!_insertManyIfNewUnit_list_of_mem_of_contains_eq_false [EquivBEq α] [LawfulHashable α]
+    [Inhabited α] (h : m.1.WF) {l : List α} {k k' : α} (k_beq : k == k')
+    (distinct : l.Pairwise (fun a b => (a == b) = false))
+    (mem : k ∈ l) : contains m k = false →
+    getKey! (insertManyIfNewUnit m l).1 k' = k := by
+  simp_to_model using getKey!_insertListIfNewUnit_of_mem_of_contains_eq_false
+
+theorem getKey!_insertManyIfNewUnit_list_mem_of_mem_of_contains [EquivBEq α] [LawfulHashable α]
+    [Inhabited α] (h : m.1.WF) {l : List α} {k : α}
+    (distinct : l.Pairwise (fun a b => (a == b) = false))
+    (h': l.contains k) : m.contains k →
+    getKey! (insertManyIfNewUnit m l).1 k = getKey! m k  := by
+  simp_to_model using getKey!_insertListIfNewUnit_of_mem_of_contains
+
+theorem getKeyD_insertManyIfNewUnit_list_of_contains_eq_false [EquivBEq α] [LawfulHashable α]
+    (h : m.1.WF) {l : List α} {k fallback : α}
+    (h' : l.contains k = false):
+    getKeyD (insertManyIfNewUnit m l).1 k fallback = getKeyD m k fallback := by
+  simp_to_model using getKeyD_insertListIfNewUnit_of_contains_eq_false
+
+theorem getKeyD_insertManyIfNewUnit_list_of_mem_of_contains_eq_false [EquivBEq α] [LawfulHashable α]
+    (h : m.1.WF) {l : List α} {k k' fallback : α} (k_beq : k == k')
+    (distinct : l.Pairwise (fun a b => (a == b) = false))
+    {mem : k ∈ l } : m.contains k = false →
+    getKeyD (insertManyIfNewUnit m l).1 k' fallback = k := by
+  simp_to_model using getKeyD_insertListIfNewUnit_of_mem_of_contains_eq_false
+
+theorem getKeyD_insertManyIfNewUnit_list_of_mem_of_contains [EquivBEq α] [LawfulHashable α]
+    (h : m.1.WF) {l : List α} {k fallback : α}
+    (distinct : l.Pairwise (fun a b => (a == b) = false))
+    (h': l.contains k) : m.contains k →
+    getKeyD (insertManyIfNewUnit m l).1 k fallback = getKeyD m k fallback := by
+  simp_to_model using getKeyD_insertListIfNewUnit_of_mem_of_contains
+
+theorem size_insertManyIfNewUnit_list [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List α}
+    (distinct : l.Pairwise (fun a b => (a == b) = false)):
+    (∀ (a : α), ¬ (m.contains a ∧ l.contains a)) →
```

Comment:
This way to state it has the advantage of being symmetic, but it has the disadvantage of not being in `simp`-normal form, as we have the `simp` lemma `not_and`. For this reason, I would suggest changing this argument to `(∀ (a : α), m.contains a → l.contains a = false)` here and everywhere else.

Reply:
I did not find many occurrences. I hope I did not miss anything.